lib/src/main/java/com/tubitv/media/player/PlayerContainer.kt

```diff
@@ -0,0 +1,278 @@
+package com.tubitv.media.player
+
+import android.content.Context
+import android.os.Handler
+import com.google.android.exoplayer2.*
+import com.google.android.exoplayer2.drm.DefaultDrmSessionManager
+import com.google.android.exoplayer2.drm.FrameworkMediaCrypto
+import com.google.android.exoplayer2.drm.FrameworkMediaDrm
+import com.google.android.exoplayer2.drm.HttpMediaDrmCallback
+import com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector
+import com.google.android.exoplayer2.upstream.DataSource
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter
+import com.google.android.exoplayer2.upstream.HttpDataSource
+import com.tubitv.media.fsm.state_machine.FsmPlayer
+import com.tubitv.media.helpers.MediaHelper
+import com.tubitv.media.models.MediaModel
+import com.tubitv.media.models.PlayerVideoResource
+import com.tubitv.media.utilities.EventLogger
+import com.tubitv.media.utilities.ExoPlayerLogger
+import java.lang.ref.WeakReference
+
+/**
+ * Central Player class that takes care of all player instance related logic
+ */
+class PlayerContainer {
+
+    companion object {
+        private val TAG = PlayerContainer::class.simpleName
+
+        private val sBandwidthMeter = DefaultBandwidthMeter()
+        private val sVideoTrackSelectionFactory = AdaptiveTrackSelection.Factory(sBandwidthMeter)
+        private val sTrackSelector = DefaultTrackSelector(sVideoTrackSelectionFactory)
+        private val sEventLogger = EventLogger(sTrackSelector)
+
+        private var sEventListener: Player.EventListener? = null // Custom listener
+
+        // Data source factory to be used for video chunk
+        private var sDataSourceFactory: DataSource.Factory? = null
+        // Data source factory without bandwidth meter
+        private var sDataSourceFactoryWithoutBandwidthMeter: DataSource.Factory? = null
+        // Data source factory to be used for DRM
+        private var sHttpDataSourceFactory: HttpDataSource.Factory? = null
+
+        private var sPlayReadyCallback: HttpMediaDrmCallback? = null
+        private var sWidevineCallback: HttpMediaDrmCallback? = null
+        private var sPlayReadyDrmSessionManager: DefaultDrmSessionManager<FrameworkMediaCrypto>? = null
+        private var sWidevineDrmSessionManager: DefaultDrmSessionManager<FrameworkMediaCrypto>? = null
+
+        private var sPlayer: SimpleExoPlayer? = null
+
+        private var sContextRef: WeakReference<Context>? = null
+        private var sHandlerRef: WeakReference<Handler>? = null
+        private var sFsmPlayerRef: WeakReference<FsmPlayer>? = null
```

Comment:
If we are keeping weakReference of Context, any some point when gc happens during the playback, Will we be able to initialized the new instance of the player for content switching?